sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/HttpPipelineAdapter.java

```diff
@@ -57,4 +84,79 @@ public IHttpResponse send(HttpRequest httpRequest) throws Exception {
                 })))
             .block();
     }
+
+    private void logAccounIdentifiersIfConfigured(String body) {
+        if (identityClientOptions != null
+            && !identityClientOptions.getIdentityLogOptionsImpl().isLoggingAccountIdentifiersAllowed()) {
+            return;
+        }
+        try {
+            JsonParser responseParser = JSON_FACTORY.createParser(body);
+            String accessToken = getTargetFieldValueFromJsonParser(responseParser, ACCESS_TOKEN_JSON_KEY);
+            responseParser.close();
+            if (accessToken != null) {
+                String[] base64Metadata = accessToken.split("\\.");
+                if (base64Metadata.length > 1) {
+                    byte[] decoded = Base64.getDecoder().decode(base64Metadata[1]);
+                    String data = new String(decoded, StandardCharsets.UTF_8);
+                    JsonParser jsonParser = JSON_FACTORY.createParser(data);
+                    HashMap<String, String> jsonMap = parseJsonIntoMap(jsonParser);
+                    jsonParser.close();
+                    String appId = jsonMap.containsKey(APPLICATION_ID_JSON_KEY)
+                        ? jsonMap.get(APPLICATION_ID_JSON_KEY) : null;
+                    String objectId = jsonMap.containsKey(OBJECT_ID_JSON_KEY)
+                        ? jsonMap.get(OBJECT_ID_JSON_KEY) : null;
+                    String tenantId = jsonMap.containsKey(TENANT_ID_JSON_KEY)
+                        ? jsonMap.get(TENANT_ID_JSON_KEY) : null;
+                    String userPrincipalName = jsonMap.containsKey(USER_PRINCIPAL_NAME_JSON_KEY)
+                        ? jsonMap.get(USER_PRINCIPAL_NAME_JSON_KEY) : null;
+
+                    CLIENT_LOGGER.log(LogLevel.INFORMATIONAL, () -> MessageFormat
+                        .format(ACCOUNT_IDENTIFIER_LOG_MESSAGE,
+                            getAccountIdentifierMessage(APPLICATION_IDENTIFIER, appId),
+                            getAccountIdentifierMessage(OBJECT_ID, objectId),
+                            getAccountIdentifierMessage(TENANT_ID, tenantId),
+                            getAccountIdentifierMessage(USER_PRINCIPAL_NAME,
+                                userPrincipalName)));
+                }
+            }
+        } catch (IOException e) {
+            CLIENT_LOGGER.log(LogLevel.WARNING, () -> "allowLoggingAccountIdentifiers Log option was set,"
+                    + " but the account information could not be logged.", e);
+        }
+    }
+
+    private String getAccountIdentifierMessage(String identifierName, String identifierValue) {
+        if (identifierValue == null) {
+            return "No " + identifierName + " available.";
```

Comment:
This is very nitpicky, but I wonder if this should just say `"none"`. Given the format of the string we'd end up with something like:
```
[Authenticated account] Client ID: No Application Identifier Available, Tenant ID:...
```

Which both feels wordy and also because the consts don't match the string you get weird disconnects.

Reply:
This we need to align with other languages too then.
For beta we keep as it is, on next meeting we can discuss this to see what other languages align with.